### ISSUE TITLE
two labels for the same input

### ISSUE THREAD

**Issue by stevemao** [evcohen/eslint-plugin-jsx-a11y#527]
coming from https://github.com/airbnb/javascript/issues/1089#issuecomment-453745514

Just ran into this issue:
I need to have two `label`s for one `input`. So the markup might be a bit messy if I have to nest input inside of two labels:
```html
<label>
  <label>
    <input />
  </label>
</label>
```
I'm not sure if this is possible but if one `input` already has a `label` as parent, the other `label` only having `htmlFor` should be enough.

`jsx-a11y/label-has-for` and `jsx-a11y/label-has-associated-control`

Basically I want clicking either label to focus the input. In my case, my first label is `Please input your username: ` the second label is an icon left to the input. This might be a rare use case though.

**Comment by ljharb** [comment]
It's an interesting case (and I suspect very rare) - I'm not sure why the icon needs to be in a label if it's already inside the outer label? Clicking anywhere in the entire box created by the label should focus the input.

**Comment by stevemao** [comment]
> I'm not sure why the icon needs to be in a label if it's already inside the outer label?

It is not already inside the outer label

They are at different sections. putting them in a same label makes the markup not semantic.

**Comment by ljharb** [comment]
It's entirely semantic to have multiple things inside the label containing the input - text, images, etc.

If you need the second label to be in a different part of the HTML, then certainly the second one would need an eslint override, and there wouldn't be a reliable way that commonly occurs enough to be worth coding that into the plugin.

Your OP shows that there's one label nested inside another:
```jsx
<label htmlFor="a">
  <label htmlFor="a">
    <input id="a" />
  </label>
</label>
```
which seems redundant, and equivalent to:
```jsx
<label htmlFor="a">
  <input id="a" />
</label>
```

**Comment by stevemao** [comment]
I simplified the markup a bit. But yeah the second label is in a different part of the HTML.

**Comment by ljharb** [comment]
In that case i think the best option is an eslint override comment on the second label.
